modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/states/WaitForFundingSigned.kt

```diff
@@ -117,29 +121,38 @@ data class WaitForFundingSigned(
-            liquidityPurchase?.let { purchase ->
-                if (channelParams.localParams.isChannelOpener) {
-                    // We only count the mining fees that we must refund to our peer as part of the liquidity purchase.
-                    // If we're also contributing to the funding transaction, the mining fees we pay for our inputs and
-                    // outputs will be recorded in the ViaNewChannel incoming payment entry below.
-                    add(ChannelAction.Storage.StoreOutgoingPayment.ViaInboundLiquidityRequest(txId = action.fundingTx.txId, localMiningFees = 0.sat, purchase = purchase))
-                    add(ChannelAction.EmitEvent(LiquidityEvents.Purchased(purchase)))
-                }
-            }
-            // If we receive funds as part of the channel creation, we will add it to our payments db.
-            if (action.commitment.localCommit.spec.toLocal > 0.msat) add(
-                ChannelAction.Storage.StoreIncomingPayment.ViaNewChannel(
-                    amountReceived = action.commitment.localCommit.spec.toLocal,
-                    serviceFee = 0.msat,
-                    // We only count the mining fees we're paying for our inputs and outputs.
-                    // The mining fees for the remote inputs and outputs are paid by the remote node.
-                    miningFee = action.fundingTx.sharedTx.tx.localFees.truncateToSatoshi(),
-                    localInputs = action.fundingTx.sharedTx.tx.localInputs.map { it.outPoint }.toSet(),
-                    txId = action.fundingTx.txId,
-                    origin = channelOrigin
+            // If we purchased liquidity as part of the channel creation, we will record it in our payments db.
+            // Only the initiator can request liquidity (the non-initiator is selling liquidity).
+            val liquidityPurchaseRequestedBySelf = if (channelParams.localParams.isChannelOpener) liquidityPurchase else null
+            liquidityPurchaseRequestedBySelf?.let { add(ChannelAction.EmitEvent(LiquidityEvents.Purchased(it))) }
+            if (action.fundingTx.sharedTx.tx.localInputs.isNotEmpty()) {
+                // If we've contributed on-chain funds to this channel, this is an incoming on-chain payment.
+                add(
+                    ChannelAction.Storage.StoreIncomingPayment.ViaNewChannel(
+                        amountReceived = action.commitment.localCommit.spec.toLocal,
+                        // The total mining fee paid is the sum of:
+                        //  - mining fees we're paying for our inputs
+                        //  - mining fees paid to the remote node for the inputs they provided as liquidity
+                        miningFee = action.fundingTx.sharedTx.tx.localFees.truncateToSatoshi() + (liquidityPurchaseRequestedBySelf?.fees?.miningFee ?: 0.sat),
+                        serviceFee = liquidityPurchaseRequestedBySelf?.fees?.serviceFee?.toMilliSatoshi() ?: 0.msat,
```

Comment:
You didn't pick this comment from my PR, I think it was useful?

```suggestion
                        // The service fee is exclusively due to the liquidity purchase. It is sort of redundant to define
                        // a separate field, but this was useful for backward compatibility (before we used liquidity ads).
                        serviceFee = liquidityPurchaseRequestedBySelf?.fees?.serviceFee?.toMilliSatoshi() ?: 0.msat,
```